test/new-e2e/tests/containers/k8s_test.go

```diff
@@ -929,6 +929,13 @@ func (suite *k8sSuite) testAdmissionControllerPod(namespace string, name string,
 		}, 5*time.Minute, 10*time.Second, "The deployment with name %s in namespace %s does not exist or does not have the auto detected languages annotation", name, namespace)
 	}
 
+	// Record old pod, so we can be sure we are not looking at the incorrect one after deletion
+	oldPods, _ := suite.K8sClient.CoreV1().Pods(namespace).List(ctx, metav1.ListOptions{
+		LabelSelector: fields.OneTermEqualSelector("app", name).String(),
+	})
+	suite.Require().Len(oldPods.Items, 1)
```

Comment:
Errors are not checked.
I suppose that the assumption was that, in case of error, `oldPods.Items` would be empty and that would be checked.
The advantage of explicitly checking the error although it might look redundant is that the error message in case of failure will be clearer and more explicit that only `Expected length of 1, got 0`.

```suggestion
	oldPods, err := suite.K8sClient.CoreV1().Pods(namespace).List(ctx, metav1.ListOptions{
		LabelSelector: fields.OneTermEqualSelector("app", name).String(),
	})
	suite.Require().NoError(err)
	suite.Require().Len(oldPods.Items, 1)
```

Reply:
That assumption was correct, but you are right in that this would definitely be more helpful for debugging